src/main/java/org/elasticsearch/search/MultiValueMode.java

```diff
@@ -274,6 +315,34 @@ protected int pick(RandomAccessOrds values) {
                 return -1;
             }
         }
+    },
+
+    /**
+     * Count of values for the field per document.
+     *
+     * Note that apply/reduce do not work with COUNT since count cannot be derived from
+     * an accumulator algorithm without a consistency between which value is the accumulation.
+     */
+    COUNT {
```

Comment:
I completely agree with this, so I moved count out of MultiValueMode and created a ValueSource/FunctionValues pair for it exclusively for expressions instead.